### PR TITLE
Create GlobalConfig

### PR DESCRIPTION
**Pull request by Adam0Sam**
If it is possible to directly access the newest values of GlobalConfig by just reading public properties, then there is no need for the `get` generic method

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/helpers/utils/GlobalConfig.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import com.acmerobotics.dashboard.config.Config;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+
+@Config
+public class GlobalConfig {
+    private static boolean DEBUG_MODE = false;
+
+    private static final Map<ConfigVariable, Object> defaultValueMap = new HashMap<>();
```
Comment (xE1H) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/helpers/utils/GlobalConfig.java]: extract all logic an abstract class, which the `GlobalConfig` one implements maybe? it doesn't really make sense to have the logic in the same class as the actual config

**Review by xE1H — CHANGES_REQUESTED** [review]
it seems a bit too overcomplicated tbh. to me it seems that a solution that involves making the class behave more like an `object` on other languages (ex. Scala) and just pulling public vars directly makes way more sense than whatever is happening here

tl;dr: turn the class into a singleton and have public vars for the actual config

**Review by xE1H — APPROVED** [review]
lgtm 🤑
